data/transactions/logic/box_test.go

```diff
@@ -300,3 +300,43 @@ func TestIOBudgetGrow(t *testing.T) {
                       int 350; byte "another"; box_create;
                       int 1`, ep)
 }
+
+func TestConveniences(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	ep, _, ledger := logic.MakeSampleEnv()
+	ledger.NewApp(basics.Address{}, 888, basics.AppParams{})
+
+	// box_get of a new name reports !exists, and returns 0 length bytes.
+	logic.TestApp(t, `byte "self"; box_get; !; assert; len; !`, ep)
+
+	// box_len of a new name reports !exists, and returns 0 as the length
+	logic.TestApp(t, `byte "self"; box_len; !; assert; !`, ep)
+
+	// box_put creates the box with contents provided
+	logic.TestApp(t, `byte "self"; byte 0x3132; box_put;
+                      byte "self"; box_len; assert; int 2; ==; assert
+                      byte "self"; box_get; assert; byte 0x3132; ==`, ep)
+
+	// box_put fails if box exists and is wrong size (self exists from last test)
+	logic.TestApp(t, `byte "self"; byte 0x313233; box_put; int 1`, ep,
```

Comment:
nit: guess you don't actually need the last `int 1`, since it would just error on `box_put`?

Reply:
My intent in these test programs is to always try to write them so that they would succeed if the error I'm looking for doesn't happen.  Without the `int 1` then this program can't possibly succeed, so  the test reduces to the string compare.  That's not all that bad, but I'd rather make is so that if the thing I'm testing changes, the behavior of the test changes drastically - in this case, if `box_put` does not fail, then this program succeeds, which means the test fails, no matter what error message I was using in the test to match against.